source/extensions/clusters/redis/redis_cluster_lb.cc

```diff
@@ -6,8 +6,17 @@ namespace Clusters {
 namespace Redis {
 
 bool ClusterSlot::operator==(const Envoy::Extensions::Clusters::Redis::ClusterSlot& rhs) const {
-  return start_ == rhs.start_ && end_ == rhs.end_ && primary_ == rhs.primary_ &&
-         replicas_ == rhs.replicas_;
+  if (start_ != rhs.start_ || end_ != rhs.end_ || *primary_ != *rhs.primary_ ||
+      replicas_.size() != rhs.replicas_.size()) {
+    return false;
+  }
+  for (auto it1 = replicas_.begin(), it2 = rhs.replicas_.begin(); it1 != replicas_.end();
+       it1++, it2++) {
+    if (**it1 != **it2) {
```

Comment:
`replicas_` is a `absl::flat_hash_set` which doesn't make any guarantees about iteration order. How can we be sure it's going to work always?

Reply:
Oh, that's a problem. how about use this code instead?
`  
  std::set<std::string> replicas_set;
  for (auto const& replica : replicas_) {
    replicas_set.emplace(replica->asString());
  }

  for (auto const& replica : rhs.replicas_) {
    if (replicas_set.find(replica->asString()) == replicas_set.end()) {
      return false;
    }
  }

  return true;
`

Reply:
I have add a new commit.

Reply:
Probably we could keep the existing code and to instantiate `replicas_` as a `absl::flat_hash_set` with a custom equality function instead?

Reply:
OK, I have understood what you said, I will try it in this way.